brave/brave-core
# implemented history clearing handler + reset leo confirmation

- Added checkboxes to clear Brave Leo Assistant for "Advanced" and "On Exit" tabs
- Implemented common handler for both cases (It is currently doing nothing).
- Added dialog to confirm when you reset Brave Leo prefs to default


Resolves https://github.com/brave/brave-browser/issues/31441



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

-  make sure that new checkboxes are visible on the `brave://settings/clearBrowserData` ("Advanced" and "On Exit" tabs) when you enabled `AI-Chat` and hidden when disabled
-  make sure that confirmation dialog works properly when you click "Clear Leo data" on the `brave://settings/leo-assistant` (It means after confirmation: `Show Leo icon in the sidebar` is ON, `Show suggested prompts in the conversation` is OFF and `AI-Chat` icon is shown on the sidebar, otherwise nothing changed)
